dds/DCPS/transport/rtps_udp/RtpsUdpDataLink.cpp

```diff
@@ -4624,8 +4639,7 @@ RtpsUdpDataLink::accumulate_addresses(const RepoId& local, const RepoId& remote,
 
   const RemoteInfoMap::const_iterator pos = locators_.find(remote);
   if (pos != locators_.end()) {
-    if (prefer_unicast && pos->second.last_recv_addr_ != NO_ADDR) {
-      normal_addrs.insert(pos->second.last_recv_addr_);
+    if (prefer_unicast && pos->second.insert_recv_addr(normal_addrs)) {
```

Comment:
I think this is probably OK, but it's worth noting that this is a change from the previous behavior, where the normal_addrs_expires and valid_last_recv_addr were always updated, regardless of the source address.

Reply:
It made sense to me to make it conditional, but if it needs to change I'm OK with that too.  The NO_ADDR case still needs the conditional.